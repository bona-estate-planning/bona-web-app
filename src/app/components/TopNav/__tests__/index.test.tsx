import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { TopNav } from '..';

describe('<TopNav  />', () => {
  const getComponent = () => (
    <BrowserRouter>
      <TopNav />
    </BrowserRouter>
  );

  it('should match snapshot', () => {
    const topNav = render(getComponent());
    expect(topNav.container.firstChild).toMatchSnapshot();
  });

  it('should contain the site title', async () => {
    render(getComponent());
    const siteTitle = await screen.getByTestId('site-title');
    expect(siteTitle).toHaveTextContent('Bona');
  });
});
