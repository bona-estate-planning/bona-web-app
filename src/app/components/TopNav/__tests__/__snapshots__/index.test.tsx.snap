// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav  /> should match snapshot 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h1
      class="MuiTypography-root makeStyles-title-1 makeStyles-title-2 MuiTypography-h6"
      data-testid="site-title"
    >
      Bona
    </h1>
    <button
      aria-label="open drawer"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
